mesonbuild/modules/python.py

```diff
@@ -263,14 +272,24 @@ def _dependency_method_impl(self, kwargs: TYPE_kwargs) -> Dependency:
         candidates = python_factory(self.interpreter.environment, for_machine, new_kwargs, self.held_object)
         dep = find_external_dependency('python', self.interpreter.environment, new_kwargs, candidates)
 
+        allow_limited_api = self.interpreter.environment.coredata.get_option(OptionKey('python.allow_limited_api'))
+        if limited_api_version != '' and allow_limited_api:
+            limited_api_version_hex = self._convert_api_version_to_py_version_hex(limited_api_version, dep.version)
+            dep.limited_api = limited_api_version
+            dep.compile_args.append(f'-DPy_LIMITED_API={limited_api_version_hex}')
+
         self.interpreter.coredata.deps[for_machine].put(identifier, dep)
         return dep
 
     @disablerIfNotFound
-    @permittedKwargs(permitted_dependency_kwargs | {'embed'})
+    @permittedKwargs(permitted_dependency_kwargs | {'embed', 'limited_api'})
     @FeatureNewKwargs('python_installation.dependency', '0.53.0', ['embed'])
+    @FeatureNewKwargs('python_installation.dependency', '1.7.0', ['limited_api'])
     @noPosargs
     def dependency_method(self, args: T.List['TYPE_var'], kwargs: 'TYPE_kwargs') -> 'Dependency':
+        if 'limited_api' not in kwargs:
+            kwargs['limited_api'] = 'inherit'
```

Comment:
Is it possible to use None as a sentinel here instead? Or did you intentionally want people to be able to do:

```
pydep = py.dependency(limited_api: 'inherit')?

Reply:
I think at some earlier point in development there was a typing issue that made it easier to use strings everywhere. I'm not seeing what that issue was anymore, so I'll go ahead and change it to `None` as the sentinel value.